.github/workflows/sims.yml

```
@@ -12,27 +16,32 @@ on:
 
 jobs:
   cleanup-runs:
-    runs-on: ubuntu-latest
+    runs-on: self-hosted
     if: "!startsWith(github.ref, 'refs/tags/') && github.ref != 'refs/heads/main'"
     steps:
       - uses: rokroskar/workflow-run-cleanup-action@master
         env:
           GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
 
   build:
-    runs-on: ubuntu-latest
+    runs-on: self-hosted
     if: "!contains(github.event.head_commit.message, 'skip-sims')"
     steps:
       - uses: actions/checkout@v3
       - uses: actions/setup-go@v4.0.0
         with:
           go-version: 1.19
+      - name: Configure git for private modules
+        run: git config --global url."ssh://git@github.com".insteadOf "https://github.com"
       - name: Display go version
         run: go version
-      - run: make build
+      - name: Build
+        run: make build
+        with:
+          ssh-key: ${{ secrets.PRIVATE_REPO_SSH_KEY }}
```

It seems that ssh configuration step is required instead of with.
```suggestion
      - name: Setup ssh config
    env:
      SSH_KEY: ${{ secrets.PRIVATE_REPO_SSH_KEY }}
    run: |
      mkdir -p ~/.ssh
      echo "$SSH_KEY" > ~/.ssh/id_rsa
      chmod 600 ~/.ssh/id_rsa
      - run: make build
```

---

result...
https://github.com/Finschia/finschia-sdk/actions/runs/5678980707/job/15390254901